### PR TITLE
s-search 0.4.2 (new formula)

### PR DESCRIPTION
**Pull request by samurai20000**
### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

_You can erase any parts of this template not applicable to your Pull Request._

### New Formulae Submissions:

- [X] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Have you built your formula locally prior to submission with `brew install <formula>`?

Fix (https://github.com/Homebrew/homebrew/pull/49328)
- Change package name from "s" to "s-search" (https://github.com/zquestz/s/issues/93)
- Become a month old

Add formula for s (https://github.com/zquestz/s), web search from the terminal. This script is written in golang.

### CODE REVIEW

File: Library/Formula/s-search.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  end
+
+  go_resource "golang.org/x/text" do
+    url "https://go.googlesource.com/text.git",
+      :revision => "07b9a78963006a15c538ec5175243979025fa7a8"
+  end
+
+  def install
+    ENV["GOPATH"] = buildpath
+    mkdir_p buildpath/"src/github.com/zquestz"
```
Comment (bfontaine) [inline Library/Formula/s-search.rb]: Use `(buildpath/"src/github.com/zquestz").mkpath`.

File: Library/Formula/s-search.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    ln_s buildpath, buildpath/"src/github.com/zquestz/s"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "-o", "s"
+    bin.install "s"
```
Comment (bfontaine) [inline Library/Formula/s-search.rb]: You can combine both lines with `system "go", "build", "-o", "#{bin}/s"`.

File: Library/Formula/s-search.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  end
+
+  test do
+    system "#{bin}/s", "--version"
```
Comment (bfontaine) [inline Library/Formula/s-search.rb]: Can’t we write a test that does something more substantial than e.g. `--help` or `--version`?

File: Library/Formula/s-search.rb (hunk 1 above)
Comment (samurai20000) [inline Library/Formula/s-search.rb]: Done!

File: Library/Formula/s-search.rb (hunk 2 above)
Comment (samurai20000) [inline Library/Formula/s-search.rb]: Done!

File: Library/Formula/s-search.rb (hunk 3 above)
Comment (samurai20000) [inline Library/Formula/s-search.rb]: No, I can't imagine more good command.

Comment (bfontaine) [inline Library/Formula/s-search.rb]: According to the documentation we can provide a browser with the `-b` option. We just need to use a dummy “browser” like `echo` and it should work e.g.:
```rb
  test do
    assert_equal "https://www.google.com/search?q=homebrew\n",
      shell_output("#{bin}/s -b echo homebrew")
  end
```

**Comment by DomT4** [comment]
Baptiste's comment about the test aside this looks great. Thanks for filing the updated PR!

File: Library/Formula/s-search.rb (hunk 3 above)
Comment (zquestz) [inline Library/Formula/s-search.rb]: That works.

```
$ s -b echo foo                            
https://www.google.com/search?q=foo
```

Comment (samurai20000) [inline Library/Formula/s-search.rb]: Thanks! I implemented the better test as you suggest.

**Comment by zquestz** [comment]
Looks like all the work is complete. @samurai20000 I have a feeling you should probably squash your commits. Then it should be ready to go. =)

**Comment by samurai20000** [comment]
@DomT4 Do I need to squash my commits?

**Comment by DomT4** [comment]
It's not mandatory but if you don't one of us has to do it locally before merging, so you'd be making our lives a little easier :)

**Comment by samurai20000** [comment]
@DomT4 Done!

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**